linked-list/hemartin/linked_list.go

```diff
@@ -0,0 +1,129 @@
+package linkedlist
+
+import (
+	"errors"
+)
+
+var ErrEmptyList error = errors.New("Error")
+
+type Node struct {
+	Val  interface{}
+	prev *Node
+	next *Node
+}
+
+type List struct {
+	Nodes []*Node
```

Comment:
You actually used a regular list (slice), while you were supposed to implement linked list. The exercise states "implement a List which holds references to the first and last node", so that struct was supposed to include 2 Node pointers, to the first and last Node.